src/Forms/Shared/Samples/Map/BrowseBuildingFloors/BrowseBuildingFloors.xaml.cs

```diff
@@ -0,0 +1,101 @@
+    public partial class BrowseBuildingFloors : ContentPage
+    {
+        private const string _portalItem = "f133a698536f44c8884ad81f80b6cfc7";
+        private FloorManager _floorManager;
+
+        // Collection of floors.
+        private readonly Dictionary<string, FloorLevel> _floorOptions = new Dictionary<string, FloorLevel>();
+
+        public BrowseBuildingFloors()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            try
+            {
+                ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+
+                // Get the portal item for a web map using its unique item id.
+                PortalItem mapItem = await PortalItem.CreateAsync(portal, _portalItem);
+                Map map = new Map(mapItem);
+
+                MyMapView.Map = map;
+
+                // Map needs to be loaded in order for floormanager to be used.
+                await MyMapView.Map.LoadAsync();
+                await MyMapView.Map.FloorManager.LoadAsync();
+                List<string> floorName = new List<string>();
+                if (MyMapView.Map.FloorManager.LoadStatus == LoadStatus.Loaded)
+                {
+                    _floorManager = MyMapView.Map.FloorManager;
+
+                    // Use the dictionary to add the level's name as the key and the FloorLevel object with the associated level's name.
+                    foreach (FloorLevel level in _floorManager.Levels)
+                    {
+                        _floorOptions.Add(level.ShortName, level);
+                        floorName.Add(level.ShortName);
+                    }
+                }
+                // Provides an error message if the floor manager failed to load.
+                else if (MyMapView.Map.FloorManager.LoadStatus == LoadStatus.FailedToLoad)
+                {
+                    await Application.Current.MainPage.DisplayAlert("Alert", "Floor manager failed to load.", "OK");
+                    return;
+                }
+
+                FloorChooser.ItemsSource = floorName;
+                FloorChooser.SelectedIndex = 0;
+            }
+            catch (Exception ex)
+            {
+                await Application.Current.MainPage.DisplayAlert("Alert", ex.Message, "OK");
+            }
+        }
+
+        private void OnFloorChooserSelectionChanged(object sender, EventArgs e)
+        {
+            // Get the name of the selected floor.
+            string selectedFloorName = FloorChooser.SelectedItem.ToString();
+
+            // Set all existing floors visibility to false.
+            foreach (FloorLevel level in _floorManager.Levels)
+            {
+                level.IsVisible = false;
+            }
+
+            // Set the selected floor visibility to true.
+            _floorOptions[selectedFloorName].IsVisible = true;
+
+            MyMapView.SetViewpoint(MyMapView.Map.InitialViewpoint);
```

Comment:
Forgot to mention, both levels and facilities can be expected to have geometries that can be used for setting the viewpoint. It would be most ideal if we could demonstrate that through this sample.